tgui/packages/tgui/interfaces/RapidConstructionDevice.tsx

```diff
@@ -0,0 +1,179 @@
+
+type Data = {
+  matterLeft: Number;
+  space: Boolean;
+  silo_upgraded: Boolean;
+  silo_enabled: Boolean;
+  root_categories: string[];
+  selected_root: string;
+  categories: Category[];
+  selected_category: string;
+  display_tabs: Boolean;
+};
+
+type Category = {
+  cat_name: string;
+  designs: Design[];
+};
+
+type Design = {
+  title: string;
+  design_id: Number;
+  selected: Boolean;
+  icon: string;
+};
+
+export const Space = (props, context) => {
+  return <span>&nbsp;</span>;
+};
+
+export const MatterItem = (props, context) => {
+  const { data } = useBackend<Data>(context);
+  const { matterLeft, space } = data;
+  return (
+    <LabeledList.Item label="Units Left">
+      {space ? <Space /> : ''}
+      {matterLeft} Units
+    </LabeledList.Item>
+  );
+};
+
+export const SiloItem = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { silo_enabled } = data;
+  return (
+    <LabeledList.Item label="Silo Link">
+      <Button.Checkbox
+        content={silo_enabled ? 'Silo Online' : 'Silo Offline'}
+        checked={silo_enabled}
+        color="transparent"
+        onClick={() => act('toggle_silo')}
+      />
+    </LabeledList.Item>
+  );
+};
+
+const CategoryItem = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { root_categories = [], selected_root } = data;
+  return (
+    <LabeledList.Item label="Category">
+      {root_categories.map((root) => (
+        <Button
+          key={root}
+          content={root}
+          selected={selected_root === root}
+          color="transparent"
+          onClick={() => act('root_category', { root_category: root })}
+        />
+      ))}
+    </LabeledList.Item>
+  );
+};
+
+const InfoSection = (props, context) => {
+  const { data } = useBackend<Data>(context);
+  const { silo_upgraded } = data;
+
+  return (
+    <Section>
+      <LabeledList>
+        <MatterItem />
+        {silo_upgraded ? <SiloItem /> : ''}
+        <CategoryItem />
+      </LabeledList>
+    </Section>
+  );
+};
+
+const DesignSection = (props, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { categories = [], selected_category } = data;
+  const [categoryName, setCategoryName] = useLocalState(
+    context,
+    'categoryName',
+    selected_category
+  );
+  const shownCategory =
+    categories.find((category) => category.cat_name === categoryName) ||
+    categories[0];
+  return (
+    <Section fill scrollable>
+      <Tabs>
+        {categories.map((category) => (
+          <Tabs.Tab
+            fluid
+            key={category.cat_name}
+            selected={category.cat_name === shownCategory.cat_name}
+            onClick={() => setCategoryName(category.cat_name)}>
+            {category.cat_name}
+          </Tabs.Tab>
+        ))}
+      </Tabs>
+      {shownCategory?.designs.map((design) => (
+        <Button
+          key={design.design_id}
```

Comment:
will `className="candystripe"` work here? I think it will make it look a little more list-y

Reply:
Not with buttons, I think.

Reply:
Alternate buttons are colored with stripes. Also the highlighting of the button disappears from alternate options if i move my cursor away from it & only the text remains highlighted

https://user-images.githubusercontent.com/110812394/205607450-cf815ef8-8455-4e50-ae8d-996ff28b07a4.mp4

Reply:
@stylemistake is the icon size alright? I coudnt find better icons for the directional window types as i just reused the radial icons for that if that isnt a problem?

Reply:
> > @Ryusenshu , @MTandi the radials are back check the video above. Any other problems/suggestions?
> 
> Yeah, sure. Just make sure to add that you can right click it to description. Beginners may never try right click to see the beginner friendly window. I noticed the Analyzer window by accident.

Got it

Reply:
I like the size of directional icons, all icons should be of that size in my opinion, and tighten up the space on buttons. Reusing the icons isn't a problem.

Reply:
I think i just might be able to tighten the spacing after all. Please hold

Reply:
Alright i have done it!!. Button spacing is at the bare minimum without clipping. The scroll bar for the glass air locks is gone!!. I cant reduce it any more than this. Is this alright? @stylemistake 

https://user-images.githubusercontent.com/110812394/205677275-ec315e6f-e0c2-490b-80d1-0d15bc54716b.mp4